modules/dnn/src/layers/einsum_layer.cpp

```diff
@@ -0,0 +1,1176 @@
+    // as opposed to "ij->"
+    mutable bool explicitEquation = false;
+    mutable bool is_parsed = false;
+
+    // Stores the subscript indices for each input in the equation
+    mutable std::vector<std::vector<int64_t>> inputSubscriptIndices;
+
+    // Keeps track of the input index of the last input that had the subscript label
+    // If the value is `-1`, it means the subscript label was never encountered or it appears in the output
+    mutable std::vector<int64_t> subscriptIndicesToLastInput;
+
+    // Holds the dimension value of the index corresponding to the subscript label
+    // `-1` indicates that the corresponding label was not encountered at all
+    mutable std::vector<int64_t> subscriptIndicesToDimValue;
+
+    // Index corresponding to each output dim corresponding to each subscript index
+    // A value of -1 means the corresponding subscript index is not found in the output
+    mutable std::vector<int64_t> subscriptIndicesToOutputIndices;
+
+    // Hold max number of alphabetic numbers
+    static const size_t numOfLetters = 52;
+
+    // Stores the count corresponding to each letter encountered
+    // A value of `0` indicates that the corresponding letter hasn't been seen at all
+    mutable std::array<int64_t, numOfLetters> letter2count;
+
+    // Hold the assigned index corresponding to the letter seen
+    // `-1` means the corresponding letter wasn't seen at all
+    mutable std::array<int64_t, numOfLetters> letter2index;
+
+    // Represents the count of unique subscript labels (subscript indices)
+    // Example 1: For the equation 'ij, jk -> ik', num_subscript_indices_ = 3 (i, j, k)
+    // Example 2: For the equation '...ij', 'jk' -> '...ik', num_subscript_indices_ = 3 (i, j, k) + number of dimensions specified by an ellipsis (across all inputs)
+    mutable int64_t numLetterIndices = 0;
+
+    // The number of dimensions that are encompassed by an "ellipsis" - "...".
+    size_t numOfEllipsisDims = 0;
+
+
+    // bool processEquation(const String& equation,  const LayerParams& params);
+    void parseEquation(const String& equation) const;
+    bool processEquation(const String& equation, const std::vector<MatShape>& inputs) const;
+    bool processBroadcastedDims() const;
+    bool createOutputSubsctipt() const;
+    bool calculateOutputShape(std::vector<MatShape>& outputDims) const;
+    bool preProcessInputs(InputArrayOfArrays& inputs);
+    Mat FinalizeOutput(const Mat& candidateOuput, const MatShape& ordered_subscript_indices_in_candidate);
+    Mat pairwiseOperandProcess(
+        const Mat& left,
+        const MatShape& leftShapeOverride,
+        const Mat& right,
+        const MatShape& rightShapeOverride,
+        const MatShape& reduceDims,
+        bool isFinalPair
+    );
+
+
+    // constructor
+    LayerEinsumImpl(const LayerParams& params)
+    {
+        setParamsFrom(params);
+        equation = params.get<String>("equation");
+        outputSize = params.get<int>("outputSize");
```

Comment:
The onnx standard require the number of Einsum layer output to be 1. I am not sure why we need to keep it as one of the params.

Reply:
I propose to remove `outputSize` field and move assert to constructor.